datahub-web-react/src/app/ingest/source/executions/IngestionSourceExecutionList.tsx

```diff
@@ -0,0 +1,163 @@
+                    executionRequestUrn: executionUrn,
+                },
+            },
+        })
+            .catch((e) => {
+                message.destroy();
+                message.error({
+                    content: `Failed to cancel execution!: \n ${e.message || ''}`,
+                    duration: 3,
+                });
+            })
+            .finally(() => {
+                message.success({
+                    content: `Successfully submitted cancellation request!`,
+                    duration: 3,
+                });
+                // Refresh once a job was cancelled.
+                setTimeout(() => onRefresh(), 2000);
+            });
+    };
+
+    const handleCancelExecution = (executionUrn: string) => {
+        Modal.confirm({
+            title: `Confirm Cancel`,
+            content:
+                'Cancelling an running execution will NOT remove any data that has already been ingested. You can use the DataHub CLI to rollback this ingestion run.',
+            onOk() {
+                onCancelExecutionRequest(executionUrn);
+            },
+            onCancel() {},
+            okText: 'Cancel',
+            cancelText: 'Close',
+            maskClosable: true,
+            closable: true,
+        });
+    };
+
+    function handleRollbackExecution(runId: string) {
+        Modal.confirm({
+            title: `Confirm Rollback`,
+            content: (
+                <div>
+                    Rolling back this ingestion run will delete any data specifically associated with with this run. If
+                    overlapping data has been ingested in previous runs, it may not be removed.
+                    <br />
+                    <br /> Are you sure you want to continue?
+                </div>
+            ),
+            onOk() {
+                message.loading('Requesting rollback...');
+                rollbackIngestion({ variables: { input: { runId } } })
+                    .then(() => {
+                        setTimeout(() => {
+                            message.destroy();
+                            refetch();
+                            onRefresh();
+                            message.success('Successfully requested ingestion rollback');
+                        }, 2000);
+                    })
+                    .catch(() => {
+                        message.error('Error requesting ingestion rollback');
+                    });
+            },
+            onCancel() {},
+            okText: 'Rollback',
+            cancelText: 'Close',
+            maskClosable: true,
+            closable: true,
+        });
+    }
+
+    const executionRequests = (data?.ingestionSource?.executions?.executionRequests as ExecutionRequest[]) || [];
+
+    return (
+        <ListContainer>
+            {!data && loading && <Message type="loading" content="Loading executions..." />}
+            {error && message.error('Failed to load executions :(')}
```

Comment:
TODO as followup: I recently changed this error message, we should make sure we update it here as well